app/src/main/java/io/homeassistant/companion/android/sensors/MobileDataManager.kt

```diff
@@ -0,0 +1,70 @@
+package io.homeassistant.companion.android.sensors
+
+import android.content.Context
+import android.provider.Settings
+import android.provider.Settings.Global.getInt
+import android.telephony.TelephonyManager
+import io.homeassistant.companion.android.R
+
+class MobileDataManager : SensorManager {
+
+    companion object {
+        val mobileDataState = SensorManager.BasicSensor(
+            "mobile_data",
+            "binary_sensor",
+            R.string.basic_sensor_name_mobile_data,
+            R.string.sensor_description_mobile_data
+        )
+        val mobileDataRoaming = SensorManager.BasicSensor(
+            "mobile_data_roaming",
+            "binary_sensor",
+            R.string.basic_sensor_name_mobile_data_roaming,
+            R.string.sensor_description_mobile_data_roaming
+        )
+    }
+
+    override val enabledByDefault: Boolean
+        get() = false
+    override val name: Int
+        get() = R.string.sensor_name_mobile_data
+
+    override val availableSensors: List<SensorManager.BasicSensor>
+        get() = listOf(mobileDataState, mobileDataRoaming)
+
+    override fun requiredPermissions(sensorId: String): Array<String> {
+        return arrayOf()
+    }
+
+    override fun requestSensorUpdate(
+        context: Context
+    ) {
+        checkState(context, mobileDataState, "mobile_data", "mdi:signal")
+        checkState(context, mobileDataRoaming, Settings.Global.DATA_ROAMING, "mdi:toggle-switch")
+    }
+
+    private fun checkState(
+        context: Context,
+        sensor: SensorManager.BasicSensor,
+        settingKey: String,
+        icon: String
+    ) {
+        if (!isEnabled(context, sensor.id))
+            return
+
+        var enabled = false
+        if (checkPermission(context, sensor.id)) {
```

Comment:
What permission are we checking here? `requiredPermissions` is blank up above.

Reply:
Removed obsolete call to checkPermission.